### PR TITLE
Update constraints (SQL and AR)

### LINKED ISSUE #24: Required fields
According to the [schema.rb](https://github.com/coredump-ch/water-sensor-api/blob/master/db/schema.rb) many fields are still nullable. Can we make the following fields non-null?

- sensors.device_name (disallow blank)
- sensors.caption (allow blank)
- sponsors.name (disallow blank)
- sponsors.active (maybe default to 't' on create)
- measurements.temperature
- measurements.sensor_id

### PR DESCRIPTION
**Pull request by schmijos**
Fixes #24

### CODE REVIEW

File: app/models/sensor.rb [hunk 1]
```diff
@@ -1,4 +1,7 @@
 class Sensor < ApplicationRecord
+  has_many :measurements, dependent: :destroy
```
Comment (dbrgn) [inline app/models/sensor.rb]: This is the `ON DELETE CASCADE` setting that fixes #46, right?

File: app/services/raw_event_reporter.rb [hunk 2]
```diff
@@ -2,6 +2,7 @@ class RawEventReporter
   def report!(event)
     measurement = parse_measurement(event.data)
     sensor = Sensor.find_or_create_by(device_name: event.coreid)
+    sensor.caption = 'LOL Sensor'
```
Comment (dbrgn) [inline app/services/raw_event_reporter.rb]: What does this class do? Does it register new incoming measurements? If yes, wouldn't this line overwrite the caption for every sensor?

File: app/models/sensor.rb (hunk 1 above)
Comment (schmijos) [reply app/models/sensor.rb]: Yes.

File: app/services/raw_event_reporter.rb (hunk 2 above)
Comment (schmijos) [reply app/services/raw_event_reporter.rb]: Good catch. That's bad.

**Comment by dbrgn** [comment]
@schmijos ok, so this is not ready for review yet, right?

**Comment by schmijos** [comment]
@dbrgn This is ready to review. I found out that the schema even is live already. I cannot reproduce the issue https://github.com/gfroerli/gfroerli-api/pull/50#issuecomment-407309216 anymore.

**Comment by dbrgn** [comment]
With this schema, can the caption be blank (empty string) or not? Or is this validation (empty or not empty) not handled by the schema at all?

**Comment by schmijos** [comment]
> With this schema, can the caption be blank (empty string) or not? Or is this validation (empty or not empty) not handled by the schema at all?

The schema allows blank strings but the model validation doesn't. As of #24 you would allow empty captions. Why?

**Comment by dbrgn** [comment]
I don't think it's necessary for the field to be required :slightly_smiling_face: But only allowing non-blank strings would be fine with me too.

**Pull request merged**